src/map/script.c

```diff
@@ -17791,10 +17788,12 @@ BUILDIN_FUNC(unitwalk)
 /// unitkill <unit_id>;
 BUILDIN_FUNC(unitkill)
 {
-	struct block_list* bl = map_id2bl(script_getnum(st,2));
+	struct block_list* bl;
 
-	if (bl != NULL)
-		status_kill(bl);
+	if(!script_rid2bl(2,bl))
+		return SCRIPT_CMD_FAILURE;
```

Comment:
You changed it from success to failure. Probably invert the if and put the status_kill back into the if